### ISSUE TITLE
SA1642: Validation incorrect for classes with generic parametes

### ISSUE THREAD

**Issue by gps7** [DotNetAnalyzers/StyleCopAnalyzers#2967]
Hi,
This case seems to be very similar to Case #415, but relates to dynamic parameters. For example StyleCop classic still finds this violation but not the analyzer. The violation of wrong type name XXX not picked up:

	`public class AuthorizationPolicy<TIDENT> `
	`{`
		`/// <summary>`
		`/// Initializes a new instance of the <see cref="AuthorizationPolicy{XXX}"/> class.`
		`/// </summary>`
		`public AuthorizationPolicy()`
		`{`
			`Id = Guid.NewGuid().ToString();`
		`}`

**Comment by sharwell** [comment]
According to the C# compiler, there is no difference between `AuthorizationPolicy{TIDENT}`, `AuthorizationPolicy{T}`, and `AuthorizationPolicy{XXX}`. All of them reference the same generic type `AuthorizationPolicy<TIDENT>`.

**Comment by gps7** [comment]
Right, the compiler will not tell, since this is a documentation mismatch with declared name. Like I said Classic StyleCop picks up this as violation.

**Comment by sharwell** [comment]
I think this would make the most sense as a new rule, as opposed to updating existing rules to start checking this.
